File: Observer Pattern/WeatherData/WeatherData.cpp
#pragma once

#include <vector>
#include "Observer/ISubject.cpp"
#include "Observer/IObserver.cpp"

using namespace std;

class WeatherData : public ISubject {
	vector<IObserver*> observers;
	float temperature;
	float humidity;
	float pressure;

public:
	WeatherData() {
		observers = vector<IObserver*>();
	}

	void registerObserver(IObserver* o) {
		observers.push_back(o);
	}

	void removeObserver(IObserver* o) {
		observers.pop_back();
	}

	void notifyObservers() {
		for (IObserver* observer: observers) {
			observer->update(this->temperature, this->humidity, this->pressure);
		}
	}

	void measurementsChanged() {
		notifyObservers();
	}

	void setMeasurements(float temperature, float humidity, float pressure) {
		this->temperature = temperature;
		this->humidity = humidity;
		this->pressure = pressure;

		measurementsChanged();
	}

	// etc methods
};

File: Patterns/Observer Pattern/CSharp/WeatherData/WeatherData.cs
using System.Collections.Generic;

namespace WeatherData
{
    public class WeatherData : Observer.ISubject 
    {
        private List<Observer.IObserver> observers;
        private float temperature;
        private float humidity;
        private float pressure;

        public WeatherData()
        {
            observers = new List<Observer.IObserver>();
        }

        public void OnRegister(Observer.IObserver o) => observers.Add(o);

        public void OnRemove(Observer.IObserver o) => observers.Remove(o);

        public void OnNotify() {
            foreach (var observer in observers) {
                observer.OnUpdate(temperature, humidity, pressure);
            }
        }

        public void MeasurementsChanged() => OnNotify();

        public void SetMeasurements(float temperature, float humidity, float pressure) {
            this.temperature = temperature;
            this.humidity = humidity;
            this.pressure = pressure;

            MeasurementsChanged();
        }

        //Etc Methods.
    }
}